examples/units_and_coordinates/radec_to_hpc_map.py

```diff
@@ -0,0 +1,134 @@
+# To convert the reference coordinate to Helioprojective we need the location of the
+# observer (i.e. where the observation was taken). We first establish the location on
+# Earth from which the observation takes place. We can convert this to a SkyCoord at the
+# observation time.
+lofar_loc = EarthLocation(lat=52.905329712*u.deg, lon=6.867996528*u.deg)
+lofar_coord = SkyCoord(lofar_loc.get_itrs(Time(obstime)))
+
+##########################################################################
+# Now we can convert the `reference_coord` to the HPC coordinate frame
+reference_coord_arcsec = reference_coord.transform_to(frames.Helioprojective(observer=lofar_coord))
+
+##########################################################################
+# Now we need to get the other parameters from the header that will be used
+# to create the new header - here we can get the cdelt1 and cdelt2 which are
+# the spatial scales of the data axes.
+cdelt1 = (np.abs(header['cdelt1'])*u.deg).to(u.arcsec)
+cdelt2 = (np.abs(header['cdelt2'])*u.deg).to(u.arcsec)
+
+##########################################################################
+# We will also required the P angle at the observation time - the image will
+# need to be rotated by this angle.
+P1 = sun.P(obstime)
+
+##########################################################################
+# Now we can use this information to create a new header using the helper
+# function `~sunpy.map.make_fitswcs_header()`. This will create a MetaDict
+# which we contain all the necessay WCS information to create a `~sunpy.map.Map`.
+# What we provide to this includes a reference coordinate (in HPC), the spatial
+# scale of the observation (i.e. `cdelt1` and `cdelt2`), and the rotation angle (P1).
+# Note that here, 1 is subtracted from the crpix1 and crpix2 values, this is because
+# the `reference_pixel` keyword in ~sunpy.map.make_fitswcs_header` is zero indexed rather
+# than the fits convention of 1 indexed.
+
+new_header = sunpy.map.make_fitswcs_header(data, reference_coord_arcsec,
+                                           reference_pixel=u.Quantity([header['crpix1']-1, header['crpix2']-1]*u.pixel),
+                                           scale=u.Quantity([cdelt1, cdelt2]*u.arcsec/u.pix),
+                                           rotation_angle=-P1,
+                                           wavelength=frequency.to(u.MHz),
+                                           observatory='LOFAR')
+
+##########################################################################
+# Let's now inspect the new header
+print(new_header)
+
+##########################################################################
+# Lets now create a `~sunpy.map.Map`
+lofar_map = sunpy.map.Map(data, new_header)
+
+##########################################################################
+# We can now plot this map and inspect it
+fig = plt.figure()
+ax = fig.add_subplot(projection=lofar_map)
+lofar_map.plot(cmap='viridis')
+lofar_map.draw_limb()
```

Comment:
Slightly off topic question, if you just load the original data/header, is sunpy clever enough draw the limb correctly on the RA-DEC grid?

Reply:
it _should_ work be if there was some sort of observer location in the header - but there is not!